### PR TITLE
replace superagent with got in taskcluster-client

### LINKED ISSUE #4455: Switch to `got` in taskcluster-client
superagent does not support streaming payloads..

### PR DESCRIPTION
**Pull request by djmitche**
Github Bug/Issue: Fixes #4455

If this is green on the first try, I will mark its changelog leve down to `patch`.

Note that there are a few scattered uses of superagent remaining in the codebase.  I'm not going to push my luck by rewriting those in this PR.

### CODE REVIEW

**Comment by djmitche** [comment]
Well, that's greener than I thought, but not green.

**Comment by djmitche** [comment]
(The issue is that it's trying to parse JSON on empty responses)

File: clients/client/src/client.js [hunk 1]
```diff
@@ -98,28 +98,27 @@ const makeRequest = exports.makeRequest = function(client, method, url, payload,
...
+    method: method.toUpperCase(),
+    agent: client._httpAgent,
+    followRedirect: false,
+    timeout: client._timeout,
+    headers: {},
+    responseType: 'text',
+    retry: 0,
+    hooks: {
+      afterResponse: [res => {
+        // parse the body, if one was given (Got's `responseType: json` fails to check content-type)
```
Comment (imbstack) [inline clients/client/src/client.js:111]: this is a good solution

**Review by imbstack — APPROVED** [review]
seems to make sense!

**Pull request merged**